tests/test_config.py

```diff
@@ -30,9 +32,21 @@ def autouse_config_tmpdir(config_tmpdir):
 def cfg(f=ALL, topdir=None):
     # Load a fresh configuration object at the given level, and return it.
     cp = configparser.ConfigParser(allow_no_value=True)
-    config.read_config(configfile=f, config=cp, topdir=topdir)
+    cp.read(config._gather_configs(f, topdir), encoding='utf-8')
```

Comment:
After searching the code, I found that `_gather_configs()` is just an internal implementation detail of `read_config()` and used only by that deprecated function. This means `_gather_configs()` is effectively deprecated too. So I don't see what this particular line change brings except some confusion, sorry.

I think you should either "upgrade" this function to the newer API if possible, or not change it at all yet if not possible and/or too much work.

Reply:
I've wrapped the call in `with pytest.deprecated_call()` as it requires a lot more work in this case. Also added a `TODO` comment.